File: backend/src/main/java/com/tjh/riskfactor/entity/form/Answer.java
package com.tjh.riskfactor.entity.form;

import lombok.Data;
import lombok.experimental.Accessors;

import com.fasterxml.jackson.annotation.JsonFormat;
import static com.fasterxml.jackson.annotation.JsonFormat.Shape;

import com.tjh.riskfactor.entity.User;

import javax.persistence.*;
import java.util.Date;
import java.util.List;
import java.util.Set;

@Data @Entity
@Table(name = "answer")
@Accessors(chain = true)
public class Answer {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id", nullable = false)
    private User creator;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id", nullable = false)
    private Task task;

    @Column(nullable = false)
    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd")
    private Date mtime;

    @OneToMany(mappedBy = "answer", cascade = CascadeType.REMOVE)
    private Set<AnswerEntry> answers;

}

File: backend/src/main/kotlin/com/tjh/riskfactor/entity/form/Answer.kt
package com.tjh.riskfactor.entity.form

import com.fasterxml.jackson.annotation.JsonFormat

import com.tjh.riskfactor.entity.IEntity
import com.tjh.riskfactor.entity.User

import java.util.Date
import javax.persistence.*

@Entity @Table(name = "answer")
class Answer(
    @get:ManyToOne
    @get:JoinColumn(nullable = false)
    var creator: User,

    @get:ManyToOne
    @get:JoinColumn(nullable = false)
    var task: Task
): IEntity() {
    @get:Column(nullable = false)
    @get:JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    var mtime: Date = Date()

    @get:Column(length = 512000)
    var body: String? = null
}
